remove use of output files

smccarthy788 opened the pull request:
@ptnapoleon, @mambocab: Please review.

Multiplexing here: https://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/314/

ptnapoleon commented on repair_tests/incremental_repair_test.py:
```diff
@@ -54,17 +53,11 @@ def sstable_marking_test(self):
-        self.assertNotIn('Repaired at: 0', output)
-
-        os.remove('sstables.txt')
+        self.assertNotIn('Repaired at: 0', [out1, out2, out3])
```
I don't think this will work?

smccarthy788 replied:
I didn't expect it to either, but it passes locally.

ptnapoleon replied:
Oh that's because it's a negative assertion. The semantics of `in` change when checking a list instead of a string. So, instead of doing substring search, we're now checking that none of `out1`, `out2`, or `out3` are literally the string `Repaired at: 0`, which they never will be, because they're making longer than that. It would be better instead to do something like
```
for out in (out1, out2, out3):
    self.assertNotIn('Repaired at: 0', out)
```

smccarthy788 commented:
Running a new multiplex here : https://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/315/

ptnapoleon commented on repair_tests/incremental_repair_test.py:
```diff
@@ -224,10 +210,7 @@ def sstable_repairedset_test(self):
-
-        os.remove('initial.txt')
-        os.remove('final.txt')
+        self.assertNotIn('Repaired at: 0', [finalOut1, finalOut2])
```
This needs to either check both of these vars, or the concatenation of them

smccarthy788 replied:
Good catch.

mambocab commented on repair_tests/incremental_repair_test.py:
```diff
@@ -177,14 +171,10 @@ def sstable_repairedset_test(self):
-            initialoutput = r.read()
-
-        matches = findall('(?<=Repaired at:).*', initialoutput)
+        matches = findall('(?<=Repaired at:).*', initialOut1 + initialOut2)
```
I'm not sure, but I think this should maybe be passed `initialOut1 + '\n' + initialOut2`, to make sure that if 1 ends with the searched-for string, and 2 starts with it, we don't accidentally treat them as one. Hopefully 1 is newline-terminated, but just in case.

mambocab replied:
Same goes for all the other instances below. If there are more than 2, or if you think it looks cleaner, you can do this easily with `'\n'.join([...])`.

mambocab commented on repair_tests/incremental_repair_test.py:
```diff
@@ -384,15 +367,10 @@ def sstable_marking_test_not_intersecting_all_ranges(self):
+        out4 = node4.run_sstablemetadata(keyspace='keyspace1').stdout
 
-        os.remove('final.txt')
+        for out in (out1, out2, out3, out4):
```
If this is the only these `out*` variables are used, you could do:

```python
for out in (node.run_sstablemetadata(keyspace='keyspace1').stdout for node in self.cluster.nodelist()):
    self.assertNotIn('Repaired at: 0', out)
```

Maybe pull that comprehension out to a list with a name. But IMO fewer intermediate variables is always good if they're only used in one place.

smccarthy788 commented:
@mambocab, I have addressed your review comments. 

Multiplexing here : https://cassci.datastax.com/view/Parameterized/job/parameterized_dtest_multiplexer/318/

mambocab commented:
Code looks good, I'm +1.

mambocab commented:
(Assuming CI is good.)
